### DISCUSSION TITLE
Node js version 16.20 --> to upgrade?

### DISCUSSION THREAD

**Discussion by worker-it** [wazuh/wazuh#21295] [Q&A] [answered]
Hi all
I've made some vulnerability checks and I realized that my Wazuh freshly installed is using Nodejs 16.20.  But the latest versions is 20 & 21...  Is wazuh is compatible with the highest versions?  If yes, how do I upgrade to one of these version?  As you can see below, node is not natively installed on my Ubuntu server so what do I have to do?  Is there a documentation on this issue somewhere?

```
:~# node
Command 'node' not found, but can be installed with:
apt install nodejs
```

```
:/usr/share/wazuh-dashboard/node/bin# ./node --version
v16.20.0
```

```
:/usr/share/wazuh-dashboard/node/bin# uname -a
Linux wazuh 5.15.0-91-generic #101-Ubuntu SMP Tue Nov 14 13:30:08 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
```

Thanks a lot in advance for your help

**Comment by TrickUK** [comment]
+1 This needs to be updated or able to be updated manually.  Could I just install the newer version of nodejs into this directory?

**Comment by AlexRuiz7** [comment, marked as answer]
Upgrading Node is not as easy as just installing the latest Node version. The whole project, (dashboards and all its plugins), is coded under that Node version. Upgrading it requires code changes to comply with Node's interfaces and features.
We are continuously trying to move to a newer Node version, but it depends on OpenSearch's pace, as we follow their project.
The latest OpenSearch version to date, 2.14.0, [uses Node 18.19.0](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/2.14.0/.nvmrc), so not even them are up to the latest.
